pkg/resourcemanager/azuresql/azuresqlmanageduser/azuresqlmanageduser_reconcile.go

```diff
@@ -61,7 +60,7 @@ func (s *AzureSqlManagedUserManager) Ensure(ctx context.Context, obj runtime.Obj
 			return false, nil
 		}
 
-		// if this is an unmarshall error - igmore and continue, otherwise report error and requeue
+		// if this is an unmarshall error - ignore and continue, otherwise report error and requeue
 		if !strings.Contains(errorString, "cannot unmarshal array into Go struct field serviceError2.details") {
```

Comment:
Is the sense of this condition correct? The comment and the **if** seem to be inverted wrt each other.

Reply:
It would be clearer if the sense of the comment was flipped to matched the if, definitely. Or if you pulled `isUnmarshalError` into a variable it probably wouldn't need a comment.

Reply:
This comment is actually across a number of different resources (code looks basically the same), and I think it's correct although possibly phrased weirdly?

It's saying: If it is an unmarshal error, continue with this reconcile, else return. That matches the code (which returns for all errors other than unmarshal).